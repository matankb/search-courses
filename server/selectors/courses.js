const { getUdacityCourses } = require("../apis/udacityCourses");
const { getKhanAcademyCourses } = require("../apis/khanAcademyCourses");
const { FilterByText } = require("../selectors/filterByText");

const getVisibleCourses = (text) => {
    
    return Promise.all([
        getUdacityCourses(),
        getKhanAcademyCourses()
    ])
    .then((courses) => {
        const formattedCourses = [].concat(...courses);
        const filterByText = new FilterByText(formattedCourses, text);
        const visibleCourses = filterByText.sortedByRelevance;

        return visibleCourses;
    })
    .catch((e) => {
        console.log(e);
    });
};

module.exports = {
    getVisibleCourses
};