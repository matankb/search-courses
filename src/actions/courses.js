import axios from "axios";

export const setCourses = (courses) => ({
    type: "SET_COURSES",
    courses
});

export const startSetCourses = () => {
    return (dispatch, getState) => {
        return axios.get("/courseslist")
            .then((response) => {
                dispatch(setCourses(response.data));
            })
            .catch((e) => {
                console.log(e);
            });
    };
};