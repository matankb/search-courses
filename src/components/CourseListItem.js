import React from "react";
import { Link } from "react-router-dom";

const CourseListItem = ({course, isShortVersion}) => {
    return (
        <div className={`course-card-wrapper ${isShortVersion ? "course-card--short" : ""}`}>
            <Link to={`/course/${course.id}`}>

                {!isShortVersion && 
                    <div className="course-card__img-container">
                        <img className="image" src={!!course.image ? `${course.image}` : "../../images/course-placeholder.png"} alt="course image placeholder"/>
                    </div>
                }

                <div className="course-card__content">
                    <h4>{course.title}</h4>
                    {!isShortVersion && 
                        <p className="show-for-desktop">{course.shortSummary}</p>
                    }

                    <div className="course-card__content__footer">
                        <div className="course-card__content__footer__data">
                            <div><p>{course.price === 0 ? "Free" : course.price}</p></div>

                            { course.duration != "" &&
                                <div className="">
                                    <p>{course.duration}</p>
                                </div>
                            }
                        </div>
                        
                        <div className="icon-container partner-logo-container">
                            <img src={`/../../images/${course.logo}`} alt="partner logo"/>
                        </div>
                    </div>
                </div>

            </Link>

        </div>
    );
};

export default CourseListItem;