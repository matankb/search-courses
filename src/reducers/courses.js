const coursesReducerDefaultState = {
    courseList: [],
    hasFoundResults: true
};

const coursesReducer = (state = coursesReducerDefaultState, action) => {
    switch(action.type) {
        case "SET_COURSES":
            return {
                ...state,
                courseList: action.courseList,
            };
        case "SET_HAS_FOUND_RESULTS":
            return {
                ...state,
                hasFoundResults: action.hasFoundResults
            }
        default: 
            return state;
    }
};

export default coursesReducer;