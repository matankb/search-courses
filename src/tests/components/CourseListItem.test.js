import React from "react";
import { shallow } from "enzyme";
import courses from "../fixtures/courses";
import CourseListItem from "../../components/CourseListItem";

test("should render the CourseListItem correctly", () => {
    const wrapper = shallow(<CourseListItem course={courses[0]}/>);
    
    expect(wrapper).toMatchSnapshot();
});