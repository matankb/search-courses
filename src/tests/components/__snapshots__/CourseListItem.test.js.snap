// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the CourseListItem correctly 1`] = `
<div>
  <h3>
    Learn react
  </h3>
  <p />
  <p>
    price: 
    300
  </p>
  <p>
    udacity
  </p>
  <a
    href="some_url"
    target="_blank"
  >
    url_to_course
  </a>
</div>
`;
